www/source/blog/2018-08-18-lift-and-shift-hab.html.md

```diff
@@ -0,0 +1,396 @@
+
+```bash
+# assumes docker and gcloud are installed locally
+$ gcloud auth configure-docker
+$ docker tag habskp/table-setting:latest eu.gcr.io/spaterson-project/table-setting:latest
+$ docker push eu.gcr.io/spaterson-project/table-setting:latest
+```
+
+Also worthwhile to comment that although GCP was chosen for this post, the same artifacts could be deployed in the same way to other Cloud provider container registries and Kubernetes clusters simply by supplying the appropriate deployment manifest files. 
+
+## Deploying to a Habitat-enabled Kubernetes cluster
+
+The below manifest deploys the Habitat managed MySQL and Table Setting images uploaded to [Google Container Registry](https://cloud.google.com/container-registry/).  Similar environment variables as before are set to configure the database:
+
+```bash
+$ cat gke-gcr-hab-core-mysql-sample.yml
+---
+apiVersion: habitat.sh/v1beta1
+kind: Habitat
+metadata:
+  name: database
+  labels:
+    app: table-setting-hab-db
+customVersion: v1beta2
+spec:
+  v1beta2:
+    image: eu.gcr.io/spaterson-project/mysql:latest
+    count: 1
+    env:
+      - name: HAB_MYSQL
+        value: "bind = '0.0.0.0'\napp_username = 'tablesetting'\napp_password = 'tablesetting'\nroot_password = 'notverysecure'"
+    service:
+      name: mysql
+      topology: standalone
+---
+apiVersion: habitat.sh/v1beta1
+kind: Habitat
+metadata:
+  name: table-setting-app
+  labels:
+    app: table-setting-hab
+customVersion: v1beta2
+spec:
+  v1beta2:
+    image: eu.gcr.io/spaterson-project/table-setting:latest
+    count: 1
+    env:
+      - name: DB_TYPE
+        value: mysql
+      - name: DB_USER
+        value: tablesetting
+      - name: DB_PASSWD
+        value: tablesetting
+    service:
+      name: table-setting
+      topology: standalone
+      bind:
+        - name: database
+          service: mysql
+          group: default
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: table-setting-lb
+spec:
+  type: LoadBalancer
+  selector:
+    habitat-name: table-setting-app
+  ports:
+  - port: 80
+    targetPort: 5000
+```
+
+As a future improvement to the above, configuration could be conveyed via Habitat's `user.toml` mechanism, more details on that [here](https://github.com/habitat-sh/habitat-operator/tree/master/examples/config). 
+
+Run the following to deploy this manifest on our Habitat enabled Kubernetes cluster:
+
+```
+$ kubectl create -f gke-gcr-hab-core-mysql-sample.yml
+habitat "database" created
+habitat "table-setting-app" created
+service "table-setting-lb" created
+```
+
+Next, wait for an external IP address to be assigned:
+
+```bash
+$ kubectl get service table-setting-lb
+NAME               TYPE           CLUSTER-IP      EXTERNAL-IP    PORT(S)        AGE
+table-setting-lb   LoadBalancer   10.51.245.180   35.189.83.69   80:30871/TCP   2m
+```
+
+Navigating to this external IP shows the expected UI:
+
+![Running with Docker Compose](media/2018-08-18-lift-and-shift-hab/hab_sql_gke_ip.png)
```

Comment:
I don't think this is the right name for the image.

Reply:
👍